### PR TITLE
Separate syntax updating from compiling

### PR DESCRIPTION
**Pull request by nittka**
This PR addresses #167. Syntax update is removed from the compilation. A separate syntax update job is introduced. The syntax updater preferences are separated from the compiler preferences.

### CODE REVIEW

File: org.elysium.parent/org.elysium.ui/src/org/elysium/ui/compiler/CompilerConsole.java [hunk 1]
```diff
@@ -102,6 +102,15 @@ public static CompilerConsole get(IFile file) {
 		return result;
 	}
 
+	/**
+	 * Returns the console for the syntax update action.
+	 */
+	public static CompilerConsole getSyntaxUpdateConsole() {
```
Comment (thSoft) [inline org.elysium.parent/org.elysium.ui/src/org/elysium/ui/compiler/CompilerConsole.java]: I know the location of these static methods is somewhat arbitrary. If you'd agree, move this method to `SyntaxUpdaterJob` and `CompilerConsole.get` to `CompilerJob`. But I don't want to bikeshed. :)

File: org.elysium.parent/org.elysium.ui/src/org/elysium/ui/compiler/CompilerConsole.java [hunk 2]
```diff
@@ -102,6 +102,15 @@ public static CompilerConsole get(IFile file) {
...
+		CompilerConsole result = ConsoleUtils.getConsole("Syntax update", FACTORY);
```
Comment (thSoft) [inline org.elysium.parent/org.elysium.ui/src/org/elysium/ui/compiler/CompilerConsole.java]: I'd suggest "Syntax updater" or "Update syntax" for the console name.

File: org.elysium.parent/org.elysium.ui/src/org/elysium/ui/compiler/SyntaxUpdaterJob.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
+package org.elysium.ui.compiler;
```
Comment (thSoft) [inline org.elysium.parent/org.elysium.ui/src/org/elysium/ui/compiler/SyntaxUpdaterJob.java]: Please move this class to `org.elysium.ui.compiler.updater`.

**Comment by nittka** [comment]
I reused the console as it is (including the static getter) because it was not a public class. Moving the job would require opening up the console API. Would this be fine with you? I would not like to duplicate the console code...

**Comment by thSoft** [comment]
> How about renaming CompilerConsole to LilyPondConsole, placing it an a "console" package and keep the static getter methods there (getCompilerConsole(IFile) and getSyntaxUpdaterConsole())?

Yes, this would be the cleanest approach. Could you please perform these refactorings?

**Comment by nittka** [comment]
Of course - will be done soon.

**Comment by nittka** [comment]
I have rebased the branch on master and made the console refactoring as discussed.

**Comment by thSoft** [comment]
Cool, thanks!

**Pull request merged**